### PR TITLE
feat(sync): Add updateSegmentWithGroundTruth

### PR DESCRIPTION
**Pull request by ragaeeb**
## Summary by CodeRabbit

- **New Features**
  - Introduced improved alignment of AI-generated tokens with ground truth text, providing more accurate word-by-word synchronization and timing estimation.
  - Added new types to represent grounded segments and tokens, including flags for unmatched words.
- **Bug Fixes**
  - Enhanced handling of punctuation, diacritics, and edge cases during token normalization and alignment.
- **Documentation**
  - Updated README example to reflect new alignment logic and usage.
- **Tests**
  - Added comprehensive tests for token synchronization, normalization, and LCS logic.
- **Chores**
  - Updated development dependencies and raised minimum required engine version.
  - Removed an unused keyword from package metadata.

### CODE REVIEW

**Comment by ragaeeb** [comment]
@codecov-commenter how can I add:
[JS Bundle Analysis](https://docs.codecov.com/docs/javascript-bundle-analysis): Save yourself from yourself by tracking and limiting bundle sizes in JS merges.

To my CI?

**Comment by ragaeeb** [comment]
@coderabbit how can I add:
[JS Bundle Analysis](https://docs.codecov.com/docs/javascript-bundle-analysis): Save yourself from yourself by tracking and limiting bundle sizes in JS merges.

To my CI?

File: src/transcript.ts [hunk 1]
```diff
@@ -349,84 +355,31 @@ export const cleanupIsolatedTokens = (markedTokens: MarkedToken[]): MarkedToken[
...
-        );
-    }
+export const updateSegmentWithGroundTruth = (segment: Segment, groundTruth: string): GroundedSegment => {
+    return {
+        end: segment.end,
+        start: segment.start,
+        text: groundTruth,
+        tokens: syncTokensWithGroundTruth(segment.tokens, groundTruth),
+    };
+};
```
[the remark this answers is not in the document]
Comment (ragaeeb) [reply src/transcript.ts:372]: Should not be undefined.

File: src/utils/transcriptUtils.ts [hunk 2]
```diff
@@ -66,33 +69,129 @@ export const isHintMatched = (tokens: Token[], hints: Hints, index: number) => {
...
+    const normalizedGTWords = groundTruthWords.map(normalizeWord);
+
+    // Step 2: Build LCS table and extract reliable "anchor" matches.
+    const lcsTable = buildLcsTable(normalizedTokens, normalizedGTWords);
+    const lcsMatches = extractLcsMatches(lcsTable, normalizedTokens, normalizedGTWords);
+
+    // Step 3: Enforce hard constraints for first and last tokens.
+    lcsMatches.set(0, 0);
+    lcsMatches.set(tokens.length - 1, groundTruthWords.length - 1); // Always anchor the last token to the last GT word
+
```
[the remark this answers is not in the document]
Comment (ragaeeb) [reply src/utils/transcriptUtils.ts]: https://github.com/ragaeeb/paragrafs/pull/5/commits/99251b3e6c877390aa847de1449a7eeb6984cc9d
